components/library/DateRangePicker/Preview/DateRangePicker.jsx

```diff
@@ -0,0 +1,214 @@
+      }));
+    }
+  }, []);
+
+  function formatDate(date) {
+    return date.toISOString().split('T')[0];
+  }
+
+  function isValidDate(date) {
+    return !isNaN(new Date(date).getDate());
+  }
+
+  return (
+    <Box>
+      <Popover
+        active={popoverActive}
+        autofocusTarget='none'
+        preferredAlignment='left'
+        preferredPosition='below'
+        fluidContent
+        sectioned={false}
+        fullHeight
+        activator={
+          <Button icon={CalendarIcon} size='slim' onClick={() => setPopoverActive(!popoverActive)}>
+            {activeDateRange.title}
+          </Button>
+        }
+        onClose={() => setPopoverActive(false)}
+      >
+        <Box
+          style={{
+            paddingTop: '16px',
+            paddingLeft: '10px',
+            paddingRight: '10px',
+            paddingBottom: '16px'
+          }}
+        >
+          <Popover.Pane fixed>
+            <InlineGrid
+              columns={{ xs: '1fr', mdDown: '1fr', md: 'max-content max-content' }}
+              gap={0}
+            >
+              <Box
+                maxWidth='212px'
+                width='100%'
+                style={{
+                  paddingRight: '20px'
+                }}
+              >
+                <Scrollable style={{ height: '334px' }}>
+                  <OptionList
+                    options={ranges.map((range) => ({
+                      value: range.title,
+                      label: range.title
+                    }))}
+                    selected={activeDateRange.title}
+                    onChange={(selected) => {
+                      const selectedRange = ranges.find((range) => range.title === selected[0]);
+                      setActiveDateRange(selectedRange);
+                    }}
+                  />
+                </Scrollable>
+              </Box>
+              <Box maxWidth='516px'>
+                <BlockStack gap='400'>
+                  <InlineStack gap='200'>
+                    <div style={{ flexGrow: 1 }}>
+                      <TextField
+                        label='Since'
+                        role='combobox'
+                        value={formatDate(activeDateRange.period.since)}
+                        onChange={(value) => handleInputValueChange('since', value)}
+                        autoComplete='off'
+                      />
+                    </div>
+                    <Icon source={ArrowRightIcon} tone='subdued' />
+                    <div style={{ flexGrow: 1 }}>
+                      <TextField
+                        label='Until'
+                        role='combobox'
+                        value={formatDate(activeDateRange.period.until)}
+                        onChange={(value) => handleInputValueChange('until', value)}
+                        autoComplete='off'
+                      />
+                    </div>
+                  </InlineStack>
+                  <div>
+                    <DatePicker
```

Comment:
Looks like the `onMonthChange` is missing here, so the arrows to change months don't currently work. Pretty small addition, there is an example in the Polaris docs that you should be able to just copy/paste to add that functionality